src/core/Akka/Actor/DeadLetterMailbox.cs

```diff
@@ -46,9 +46,13 @@ public DeadLetterMessageQueue(IActorRef deadLetters)
         /// <param name="envelope">TBD</param>
         public void Enqueue(IActorRef receiver, Envelope envelope)
         {
-            if (envelope.Message is DeadLetter)
+            if (envelope.Message is AllDeadLetters)
```

Comment:
Important fix - now that we can produce `SupressedDeadLetter`s more easily, this safety check has to be expanded to encompass `AllDeadLetters` and their derived types, otherwise this will produce a `StackOverflowException`.